### PR TITLE
[spirv] Support Status arg in Sample methods.

### PR DESCRIPTION
**Pull request by ehsannas**

### CODE REVIEW

File: tools/clang/include/clang/SPIRV/ModuleBuilder.h [hunk 1]
```diff
@@ -171,17 +171,21 @@ class ModuleBuilder {
   /// OpImageSample* will be generated; otherwise, *ImplicitLod variant will
   /// be generated.
   ///
-  /// If bias, lod, or grad is given a non-zero value, an additional image
-  /// operands, Bias, Lod, or Grad, will be attached to the current instruction,
-  /// respectively.
+  /// If bias, lod, grad, or minLod is given a non-zero value, an additional
+  /// image operands, Bias, Lod, Grad, or minLod will be attached to the current
```
Comment (antiagainst) [inline tools/clang/include/clang/SPIRV/ModuleBuilder.h]: s/minLod/MinLod/

File: tools/clang/include/clang/SPIRV/InstBuilder.h [hunk 2]
```diff
@@ -858,6 +858,13 @@ class InstBuilder {
   InstBuilder &opConstant(uint32_t result_type, uint32_t result_id,
                           uint32_t value);
 
+  // All-in-one method for creating different types of OpImageSample*.
+  InstBuilder &
+  opImageSample(uint32_t result_type, uint32_t result_id,
+                uint32_t sampled_image, uint32_t coordinate, uint32_t dref,
+                llvm::Optional<spv::ImageOperandsMask> image_operands,
+                bool isExplicit, bool isSparse);
```
Comment (antiagainst) [inline tools/clang/include/clang/SPIRV/InstBuilder.h]: Would be nice to keep the same naming convention for the same signature.

File: tools/clang/lib/SPIRV/InstBuilderManual.cpp [hunk 3]
```diff
@@ -80,6 +80,38 @@ InstBuilder &InstBuilder::opConstant(uint32_t resultType, uint32_t resultId,
...
 }
 
+InstBuilder &InstBuilder::opImageSample(
+    uint32_t result_type, uint32_t result_id, uint32_t sampled_image,
+    uint32_t coordinate, uint32_t dref,
+    llvm::Optional<spv::ImageOperandsMask> image_operands, bool isExplicit,
+    bool isSparse) {
+  spv::Op op = spv::Op::Max;
+  if (dref) {
+    op = isExplicit ? (isSparse ? spv::Op::OpImageSparseSampleDrefExplicitLod
```
Comment (antiagainst) [inline tools/clang/lib/SPIRV/InstBuilderManual.cpp]: Nice!

File: tools/clang/include/clang/SPIRV/InstBuilder.h (hunk 2 above)
Comment (ehsannas) [reply tools/clang/include/clang/SPIRV/InstBuilder.h]: Oops. Will fix in the next CL.

File: tools/clang/include/clang/SPIRV/ModuleBuilder.h (hunk 1 above)
Comment (ehsannas) [reply tools/clang/include/clang/SPIRV/ModuleBuilder.h]: same

**Pull request merged**